src/Microsoft.Azure.ServiceBus/Primitives/ManagedServiceIdentityTokenProvider.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.ServiceBus.Primitives
+{
+    using System;
+    using System.Threading.Tasks;
+    using Azure.Services.AppAuthentication;
+
+    /// <summary>
+    /// Represents the Azure Active Directory token provider for Azure Managed Service Identity integration.
+    /// </summary>
+    public class ManagedServiceIdentityTokenProvider : TokenProvider
+    {
+        static AzureServiceTokenProvider azureServiceTokenProvider = new AzureServiceTokenProvider();
+
+        /// <summary>
+        /// Gets a <see cref="SecurityToken"/> for the given audience and duration.
+        /// </summary>
+        /// <param name="appliesTo">The URI which the access token applies to</param>
+        /// <param name="timeout">The time span that specifies the timeout value for the message that gets the security token</param>
+        /// <returns><see cref="SecurityToken"/></returns>
+        public async override Task<SecurityToken> GetTokenAsync(string appliesTo, TimeSpan timeout)
+        {
+            string accessToken = await azureServiceTokenProvider.GetAccessTokenAsync(Constants.AadServiceBusAudience);
```

Comment:
>GetAccessTokenAsync [](start = 65, length = 19)

this thing doesn't take timeout? or does it have a default timeout?

Reply:
Unfortunately not but we can expect this in the non-preview client.

---
In reply to: [156584184](https://github.com/Azure/azure-service-bus-dotnet/pull/385#discussion_r156584184) [](ancestors = 156584184)